system/jhtree/jhtree.cpp

```diff
@@ -2339,33 +2319,34 @@ extern jhtree_decl void getNodeCacheInfo(ICacheInfoRecorder &cacheInfo)
 void CNodeCache::getCacheInfo(ICacheInfoRecorder &cacheInfo)
 {
     CriticalBlock block(lock);
-    preloadCache.reportEntries(cacheInfo); // Debatable whether we should include this
     blobCache.reportEntries(cacheInfo);
     nodeCache.reportEntries(cacheInfo);
     leafCache.reportEntries(cacheInfo);
 }
 
-CJHTreeNode *CNodeCache::getNode(INodeLoader *keyIndex, unsigned iD, offset_t pos, IContextLogger *ctx, bool isTLK)
+CJHTreeNode *CNodeCache::getNode(INodeLoader *keyIndex, unsigned iD, offset_t pos, NodeType type, IContextLogger *ctx, bool isTLK)
 {
     // MORE - could probably be improved - I think having the cache template separate is not helping us here
     // Also one cache per key would surely be faster, and could still use a global total
     if (!pos)
         return NULL;
-    { 
-        // It's a shame that we don't know the type before we read it. But probably not that big a deal
-        CriticalBlock block(lock);
-        CKeyIdAndPos key(iD, pos);
-        if (preloadNodes)
-        {
-            CJHTreeNode *cacheNode = preloadCache.query(key);
-            if (cacheNode)
-            {
-                cacheHits++;
-                if (ctx) ctx->noteStatistic(StNumPreloadCacheHits, 1);
-                preloadCacheHits++;
-                return LINK(cacheNode);
-            }
-        }
+
+    CKeyIdAndPos key(iD, pos);
+    NodeType cacheType = isTLK ? NodeBranch : type;
```

Comment:
I am assuming you want to keep this as it's the way it was in old code, but I'm not sure it's especially logical. If anything TLK leaves should have their own cache (and thus their own limits on how many prepared to cache)